release.py

```diff
@@ -415,15 +415,19 @@ def cf_hash(opts, pargs):
 if __name__ == "__main__":
     parser = create_option_parser()
     (opts, pargs) = parser.parse_args()
-    
+
     if len(pargs) < 2:
         parser.error("Improper usage. Too few arguments!")
     if len(pargs) > 2:
         parser.error("Improper usage. Too many arguments!")
-    
+
+    # Prevent release and pre_release both being true
+    if opts.release and opts.pre_release:
+        parser.error("Cannot release and pre-release at the same time.")
```

Comment:
in general I like "reason for error. what to do to fix it" as a recipe that is most useful for users.  So adding "please rerun specifying one or the other" would work.  Here it is kind of obvious so not strictly necessary, but just good to get into the habit.  When error messages are aimed at a user, try and put yourself in the mind of the user, who in general is not a developer so avoid jargon.   Just some general comments there.....what you did was really great.

Reply:
Yes. I will write error messages from the perspective of "reason for error. what to do to fix it".